core/store/src/trie/mod.rs

```diff
@@ -687,13 +729,37 @@ impl Trie {
         &self,
         f: &mut dyn std::io::Write,
         hash: &CryptoHash,
-        max_depth: u32,
         spaces: &mut String,
         prefix: &mut Vec<u8>,
+        max_depth: u32,
+        limit: &mut u32,
+        type_str: Option<&str>,
+        from: Option<&[u8]>,
+        to: Option<&[u8]>,
     ) -> std::io::Result<()> {
-        if max_depth == 0 {
+        if max_depth == 0 || *limit == 0 {
             return Ok(());
         }
+        *limit -= 1;
+
+        fn is_lex_greater_than_or_prefix_of(v1: &[u8], v2: &[u8]) -> bool {
+            if v1.len() <= v2.len() && v2.starts_with(v1) {
+                return true;
+            }
+            v1 > v2
```

Comment:
Why not just use the built in comparison instead of this method?

Reply:
If the current key is a prefix of `to`, it is lexicographically smaller than `to`, yet we do not want to prune its subtree (we would prune `to`).